### DISCUSSION TITLE
How to redirect IO?

### DISCUSSION THREAD

**Discussion by guyutongxue** [pyodide/pyodide#2080] [Q&A] [answered]
By default, `input` in python code will cause a `prompt` in the browser, and `print` will be a `console.log`. I want to redirect these I/O to other places.

There is a `pyodide.setStandardStreams` in [the documentation](https://pyodide.org/en/stable/usage/api/js-api.html#pyodide.setStandardStreams), but it doesn't actually exist. I've also found that in https://github.com/pyodide/pyodide/pull/1936#issuecomment-963891899, this API will be removed later. 

Is there any other solution? Thanks a lot.

**Comment by ryanking13** [comment, marked as answer]
You can override `stdin`, `stdout`, `stderr` behaviors when calling `pyodide.loadPyodide`.
See: https://pyodide.org/en/stable/usage/faq.html#how-can-i-control-the-behavior-of-stdin-stdout-stderr

Note that `pyodide.setStandardStreams` is used internally in `pyodide.loadPyodide`.

**Reply by guyutongxue** [reply]
I got it! Thank you!

Note: From my side, the official website does not seem to be up to date, I'm viewing the source code [here](https://github.com/pyodide/pyodide/blob/ce2746b690b5f2e93d7e7b12054a5abf67cfdd1b/docs/usage/faq.md).
